tests/unit/generate_detect_target_contour.py

```diff
@@ -0,0 +1,192 @@
+"""
+Helper functions for `test_detect_target_contour.py`.
+
+"""
+
+import cv2
+import math
+import numpy as np
+
+
+LANDING_PAD_COLOUR_BLUE = (100, 50, 50)  # BGR
+
+
+# Test functions use test fixture signature names and access class privates
+# No enable
+# pylint: disable=protected-access,redefined-outer-name
+
+
+class LandingPadImageConfig:
+    """
+    Represents the data required to define and generate a landing pad.
+    """
+    def __init__(
+        self,
+        center: tuple[int, int],
+        axis: tuple[int, int],
+        blur: bool,
+        angle: float,
+    ):
+        """
+
+        center: The (x, y) coordinates representing the center of the landing pad.
+        axis: The pixel lengths of the semi-major and semi-minor axes of the ellipse.
+        blur: Indicates whether the landing pad should have a blur effect.
+        angle: The rotation angle of the landing pad in degrees clockwise, where 0.0 degrees
+            is where both semi major and minor are aligned with the x and y-axis respectively (0.0 <= angle <= 360.0).
```

Comment:
Mention only the semi-major axis, since the orientation of the semi-minor axis is a derived value.